### PR TITLE
Removes various helpers that can be simplified in a single variable check

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do
Removes helpers that can be easily simplified into a single variable check. Current list:
- `is_sharp()`
- `get_turf_or_move()` 
- `is_dead()` - This one is removed because the check `== DEAD` is used far more than the proc
- `hasorgans()`

## Why It's Good For The Game
These can just be simplified and are needlessly using helper procs

## Testing
Compiled, sliced some food

## Changelog
NPFC

### CODE REVIEW

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -1124,32 +1124,20 @@ Returns 1 if the chain up to the area contains the given typepath
...
 	if(is_pointed(W))
-		return 1
+		return TRUE
 
 	if(is_type_in_typecache(W, GLOB.can_embed_types))
-		return 1
-
-//Whether or not the given item counts as sharp in terms of dealing damage
-/proc/is_sharp(obj/O)
-	if(!O)
```
Comment (farie82) [inline code/__HELPERS/unsorted.dm:1148]: Be sure that the null check here is either also done, or isn't needed

Comment (DGamerL) [reply code/__HELPERS/unsorted.dm:1148]: After looking into it, pretty much every use was in some place that if it was done with no item it'd runtime, like `attackby`. The only place I see it maybe having issues is `adjustBruteLossByPart` so I'll test if that works, if not, I'll add a nullcheck

**Review by Henri215 — APPROVED** [review]
LGTM

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -970,7 +970,7 @@
 
 /mob/living/proc/attempt_harvest(obj/item/I, mob/user)
 	if(user.a_intent == INTENT_HARM && stat == DEAD && butcher_results) //can we butcher it?
-		var/sharpness = is_sharp(I)
+		var/sharpness = I.sharp
 		if(sharpness)
```
Comment (Contrabang) [inline code/modules/mob/living/living.dm:974]:
```suggestion
		if(I.sharp)
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
small thing, looks pretty good

**Review by Contrabang — APPROVED** [review]
Lets do this

**Pull request merged**